src/lib/field-next/_core.shape.scss

```diff
@@ -0,0 +1,20 @@
+@use '../core/styles/shape';
+@use './token-utils' as *;
+
+@mixin density($density, $inset: false) {
+  @if $inset {
+    @include override('container-block-size', #{token(inset-height-#{$density}, custom)}, value);
+  } @else {
+    @include override('container-block-size', #{token(height-#{$density}, custom)}, value);
+  }
+}
+
+@mixin shape($shape) {
+  @if $shape == 'squared' {
+    @include override('shape', 0, value);
+  } @else if $shape == 'rounded' {
+    @include override('shape', shape.variable(full), value);
+    @include override('container-padding-start', 'pill-container-padding-start');
+    @include override('container-padding-end', 'pill-container-padding-end');
```

Comment:
We may want to introduce some inline padding in the rounded variant to align the content with the half circle.